packages/passwordless-auth-construct/src/custom-auth/custom_auth_service.ts

```diff
@@ -210,9 +219,48 @@ export class CustomAuthService {
 
     const method = this.validateSignInMethod(signInMethod);
 
-    return this.challengeServiceFactory
+    const verifyResult = this.challengeServiceFactory
       .getService(method)
       .verifyChallenge(event);
+
+    const answerCorrect = (await verifyResult).response.answerCorrect;
+    logger.debug(`Answer is correct: ${answerCorrect ? 'true' : 'false'}`);
+
+    let passwordlessConfiguration: Record<string, string> | undefined;
+
+    try {
+      passwordlessConfiguration =
+        event.request.userAttributes[PASSWORDLESS_SIGN_UP_ATTR_NAME] &&
+        JSON.parse(
+          event.request.userAttributes[PASSWORDLESS_SIGN_UP_ATTR_NAME]
+        );
+    } catch (_err) {
```

Comment:
There are a few places where this pattern is used. Would be better to make a utility function like `tryParseJson` that returns undefined instead of throwing if a string is not parseable.

Reply:
Can we discuss that in a follow up PR, just to verify if other packages needs this as well